### DISCUSSION TITLE
portrange parameter

### DISCUSSION THREAD

**Discussion by chenhayat** [crossbario/crossbar#2111] [General]
Hi,
I am using in config.yaml in the portrange parameter

> transports:
>   - type: web
>     endpoint:
>       type: tcp
>       portrange: [1024,65535]

which works well.
Now, I am now trying to pass a parameter from outside of start_port_range and end_port_range but I get errors when it's trying to convert it regarding the string to int conversion.
I also tried to use it as following:

>        portrange: !!seq
>       - !!int {{ PORT_RANGE_START }}
>       - !!int {{ PORT_RANGE_END }}
> 

but got error: yaml.constructor.ConstructorError: expected a scalar node, but found mapping
I also tried to use the !template header but it does not know it.
Do you have any idea how I can pass a parameter here?

**Comment by oberstet** [comment]
`portrange` must be a sequence of two integers, this is checked here:

https://github.com/crossbario/crossbar/blob/0089c1ef6fbbb87fc7316088a91f1859fa84eeb0/crossbar/common/checkconfig.py#L988

and the integers must be valid IP ports, which is checked via the helper here:

https://github.com/crossbario/crossbar/blob/0089c1ef6fbbb87fc7316088a91f1859fa84eeb0/crossbar/common/checkconfig.py#L865

**Comment by chenhayat** [comment]
Ok. Thank you!
